runtime/Jsx.res

```diff
@@ -38,3 +38,5 @@ type component<'props> = componentLike<'props, element>
 
 /* this function exists to prepare for making `component` abstract */
 external component: componentLike<'props, element> => component<'props> = "%identity"
+
+external asyncComponent: promise<element> => element = "%identity"
```

Comment:
This is in rescript-react as React.promise(element):

https://github.com/rescript-lang/rescript-react/blob/c8c3d05724e767a3bfe22173dcf3e98140e1a6ca/src/React.res#L8C10-L8C26

I think we should also call it `promise` here.